From b73b2fc3b952064d635811450c2813f630fb3d2a Mon Sep 17 00:00:00 2001
From: Meng Wei <wei.meng@intel.com>
Date: Mon, 3 Dec 2018 16:53:09 +0800
Subject: [PATCH 0989/1214] media: intel-ipu4: fix 2 kernel panic in ipu driver

fix 2 kernel panic in ipu driver:
    1, double free
    2, memcpy after allocation failed.

Change-Id: I59597b1eb98919a29276729b51836755ae807278
Tracked-On: PKT-1732
Signed-off-by: Zhang Ning <ning.a.zhang@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/pci/intel/ipu.c b/drivers/media/pci/intel/ipu.c
index b235de1..39dfb7b 100644
--- a/drivers/media/pci/intel/ipu.c
+++ b/drivers/media/pci/intel/ipu.c
@@ -334,6 +334,10 @@ int request_cpd_fw(const struct firmware **firmware_p, const char *name,
 			return -ENOMEM;
 		tmp->size = fw->size;
 		tmp->data = vmalloc(fw->size);
+		if (!tmp->data) {
+			kfree(tmp);
+			return -ENOMEM;
+		}
 		memcpy((void *)tmp->data, fw->data, fw->size);
 		*firmware_p = tmp;
 		release_firmware(fw);
-- 
2.7.4

